Soupway/src/bar/SoupSeller.java

```diff
@@ -0,0 +1,92 @@
+package bar;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import constants.Bread;
+import constants.IceCream;
+import cookFactory.AbstractCookFactory;
+import cookFactory.CookFactoryManager;
+import menu.Soup;
+import orders.CustomerInteractor;
+import orders.Order;
+import orders.OrderBuilder;
+
+public class SoupSeller {
+	private String name;
+	private CustomerInteractor customerInteractor;
+
+	public SoupSeller(String name) {
+		this.name = name;
+		this.customerInteractor = CustomerInteractor.getInstance();
+	}
+
+	public Order processCustomerOrder() {
+		System.out.println(String.format("Hello, My name is %s. Are you vegetarian? (Y/n)", name));
+
+		boolean isVegetarian = customerInteractor.getCustomerBoolenInput();
+
+		AbstractCookFactory abstractFactory = CookFactoryManager.getFactory(isVegetarian);
+
+		OrderBuilder orderBuilder = placeOrder(abstractFactory);
+
+		return orderBuilder.build();
+	}
+
+	private OrderBuilder placeOrder(AbstractCookFactory abstractFactory) {
+		OrderBuilder orderBuilder = new OrderBuilder();
+		
+		System.out.println("I am preparing your soup...");
+		Soup orderedSoup = abstractFactory.cookSoup();
+
+		List<Enum> breadAndIcecream = new ArrayList<>();
+		
+		System.out.println("Do you want bread? (Y/n)");
+		Enum<Bread> orderedBread = null;
+		boolean wannaBread = customerInteractor.getCustomerBoolenInput();
+		if (wannaBread) {
+			orderedBread = abstractFactory.serveBread();
```

Comment:
Here you can directly call the builder like this
```
if (wannaBread) {
    orderBuilder.add(abstractFactory.serveBread())
}
```

Instead of having ``` Enum<Bread> orderedBread = null; ```